compaction/compaction_manager.cc

```diff
@@ -191,6 +191,13 @@ void compaction_manager::deregister_weight(int weight) {
     reevaluate_postponed_compactions();
 }
 
+std::vector<sstables::shared_sstable> in_strategy_sstables(compaction::table_state& table_s) {
```

Comment:
The patch title says `compaction:` but it is moved to compaction_manager.

I wonder if this series could be a good opportunity to also rename table_state to compaction_group.

Reply:
> I wonder if this series could be a good opportunity to also rename table_state to compaction_group.

I also thought of this. As we change the relationship of `table_state <-> table` from `1 <-> 1` to `N <-> 1`, the `compaction_group` name makes much more sense.

Reply:
Another (albeit minor) advantage is that it further detaches compaction from table, even if in name only.

Reply:
One more thing to consider for keeping table_state for now, though I don't like it and I'd like a better abstraction -
is TWCS double-decker design, where a compaction strategy is asked to compact a TWCS window during normal operation.  To make the two strategies composable (with STCS in master and ICS in enterprise), I think it would make sense to call `compaction_manager::submit` with an abstraction of the TWCS window.  This is sort of a compaction sub-group.  Today is a subset of the table's sstables and in the future it will be a subset of the compaction group sstables.
So this abstraction should map well to both cases and should point back to a compaction_group and the latter should point back to the table.

Reply:
Each compaction group has a compaction policy which is inherited from the underlying replica::table. Alternatively, TWCS can just abstract the policy performed on each bucket, so upper layers don't have to care about it nor implement bucket isolation. With TWCS itself being aware of all buckets, cross-bucket compaction is simpler too, which happens with major.

Reply:
Cc @cmm